### DISCUSSION TITLE
Detect lost mariadb connection (Connection was killed)

### DISCUSSION THREAD

**Discussion by ttrig** [laravel/framework#42192] [Ideas]
Hello.

I run octane with a database connection with multiple mysql hosts configured.
When a mysql host goes down, everything still works thanks to the [DetectsLostConnections](https://github.com/laravel/framework/blob/9.x/src/Illuminate/Database/DetectsLostConnections.php) trait (`server has gone away`) :tada: 
But when switching to mariadb the error message is `SQLSTATE[70100]: <<Unknown error>>: 1927 Connection was killed` and is not caught by the DetectsLostConnections trait.

Can we add a part of the mariadb message or am i doing something wrong?

I can create a PR but i dont know if to use the full message or a part of it.

https://mariadb.com/kb/en/mariadb-error-codes/

**Comment by FelipeVeiga** [comment]
same here
